jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/BlockingTest.java

```diff
@@ -36,108 +34,84 @@ public class BlockingTest
     @Test
     public void testRunBlock() throws Exception
     {
-        long start;
         try (Blocker.Runnable runnable = Blocker.runnable())
         {
             runnable.run();
-            start = NanoTime.now();
             runnable.block();
         }
-        assertThat(NanoTime.millisSince(start), lessThan(500L));
     }
 
     @Test
     public void testBlockRun() throws Exception
     {
-        long start;
         try (Blocker.Runnable runnable = Blocker.runnable())
         {
-            final CountDownLatch latch = new CountDownLatch(1);
-
+            CyclicBarrier barrier = new CyclicBarrier(2);
             new Thread(() ->
             {
-                latch.countDown();
                 try
                 {
-                    TimeUnit.MILLISECONDS.sleep(100);
+                    barrier.await(5, TimeUnit.SECONDS);
+                    runnable.run();
```

Comment:
@lorban suggested to wait for the main thread to be non-`RUNNABLE`, then call `run()`.

Reply:
Let me rework the tests to make the spawned thread wait on the blocked thread' state.